CRM/Sepa/Logic/Queue/Close.php

```diff
@@ -22,285 +22,402 @@
  */
 class CRM_Sepa_Logic_Queue_Close {
 
-  public $title               = NULL;
-  protected $txgroup          = NULL;
-  protected $target_status_id = NULL;
-  protected $group_name       = NULL;
-  protected $counter          = NULL;
+  /**
+   * Create a task for the queue
+   *
+   * @param $mode string
+   * @param $params array
+   * @return CRM_Queue_Task
+   */
+  private static function createTask($mode, $params = []) {
+    $task = new CRM_Queue_Task(
+      ['CRM_Sepa_Logic_Queue_Close', 'run'],
+      [$mode, $params],
+      self::taskTitle($mode, $params)
+    );
+
+    $task->runAs = [
+      'contactId' => CRM_Core_Session::getLoggedInContactID(),
+      'domainId'  => 1,
+    ];
+
+    return $task;
+  }
 
   /**
-   * Use CRM_Queue_Runner to close a SDD group
-   * This doesn't return, but redirects to the runner
+   * Use Civi::queue to close a SDD group
    */
   public static function launchCloseRunner($txgroup_ids, $target_group_status, $target_contribution_status) {
-    // create a queue
-    $queue = CRM_Queue_Service::singleton()->create(array(
-      'type'  => 'Sql',
-      'name'  => 'sdd_close',
-      'reset' => TRUE,
-    ));
-
-    // fetch the groups
-    $txgroup_query = civicrm_api3('SepaTransactionGroup', 'get', array(
-      'id'           => array('IN' => $txgroup_ids),
-      'option.limit' => 0
-      ));
-
-    $group_status_id_busy = (int) CRM_Core_PseudoConstant::getKey('CRM_Batch_BAO_Batch', 'status_id', 'Data Entry');
+    // Create a queue
+    $queue = Civi::queue('sdd_close', [
+      'error'  => 'abort',
+      'reset'  => TRUE,
+      'runner' => 'task',
+      'type'   => 'SqlParallel',
```

Comment:
Thought about it a bit more, and `SqlParallel` doesn't make sense with the current implementation. The queries in `updateContributions()` do not use `OFFSET`, just `LIMIT`. If we run multiple items in parallel, they'll just start with the same 250 contributions and race each other. Let's stick with sequential `Sql` for now.